paper_search_mcp/server.py

```
@@ -24,11 +25,67 @@
+
+
+SELECTED_TRANSPORT = _determine_transport()
+
+
+def _determine_host(transport: str) -> str:
+    if explicit_host := os.environ.get("PAPER_SEARCH_MCP_HOST"):
+        return explicit_host
+    if explicit_host := os.environ.get("HOST"):
+        return explicit_host
+    if transport == "streamable-http":
+        return "0.0.0.0"
```

Ruff (0.12.2)

Possible binding to all interfaces

(S104)